lib/teen_process.js

```diff
@@ -48,22 +49,33 @@ function exec (cmd, args = [], opts = {}) {
     // keep track of stdout/stderr if we haven't said not to
     if (!opts.ignoreOutput) {
       proc.stdout.on('data', (data) => {
-        stdout += data;
+        stdoutArr.push(data);
       });
       proc.stderr.on('data', (data) => {
-        stderr += data;
+        stderrArr.push(data);
       });
     }
 
+    function getStdio (isBuffer) {
+      let stdout, stderr;
```

Comment:
I think this might be shorter

```js
let stdout = Buffer.concat(stdoutArr);
let stderr = Buffer.concat(stderrArr);
if (!isBuffer) {
   stdout = stdout.toString(opts.encoding);
   stderr = stderr.toString(opts.encoding);
}
return {stdout, stderr};
```